php

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# Usage: php [options] [path-to-php-script]
+# Example: php -a
+set -eu
+
+dir="$(dirname "$0")"
+cd "$dir"
+
+. .env
+
+# note: port 9000 should match XDEBUG_REMOTE_PORT in docker-compose.yml
+# note: that port must be open on the local machine to allow debugging!
+# todo: manage the port via .env, just like the address
+# todo: once docker-compose supports -w options, set the working dir to /var/www/mediawiki (or take the from .env).
+docker-compose exec "web" php -dxdebug.remote_enable=1 -dxdebug.remote_mode=req -dxdebug.remote_port=9000 -dxdebug.remote_host=$IDELOCALHOST "$@"
```

Comment:
So some of these setting should already be set in the web container due to the env vars that are set in the docker-compose file.

     - PHP_DEBUGGER=xdebug
     - XDEBUG_REMOTE_AUTOSTART=1
     - XDEBUG_REMOTE_HOST=${IDELOCALHOST}
     - XDEBUG_REMOTE_PORT=9000
     - XDEBUG_REMOTE_CONNECT_BACK=0
     - PHP_IDE_CONFIG=serverName=docker

Did these not work for you?
I notice remote_enable isn't listed there, nor remote_mode.
But debugging phpunit tests for example using the phpunit-file script in this repo has always worked for me.
Perhaps not for you though?

Reply:
I just tried again, and the one parameter that seems to be needed is -dxdebug.remote_host=$IDELOCALHOST. Without that, it does not work.

Semi-related: one thing that should probably be documented is the fact that firewalls like ufw interfere with this. I have not yet found a good way to tell it to allow connections coming out of docker containers to local services, without specifying the current public IP of the host.

Reply:
maybe relevant: https://stackoverflow.com/questions/43326154/xdebug-inside-docker-container-ignoring-xdebug-config-environment-variable